zng/resolver/context.go

```diff
@@ -202,17 +202,36 @@ func (c *Context) AddType(t zng.Type) zng.Type {
 // and returned.  The closure of types within the columns must all be from
 // this type context.  If you want to use columns from a different type context,
 // use TranslateTypeRecord.
-func (c *Context) LookupTypeRecord(columns []zng.Column) *zng.TypeRecord {
+func (c *Context) LookupTypeRecord(columns []zng.Column) (*zng.TypeRecord, error) {
+	// First check for duplicate columns
+	names := make(map[string]struct{})
+	var val struct{}
+	for _, col := range columns {
+		_, exists := names[col.Name]
+		if exists {
```

Comment:
```suggestion
		if _, ok := names[col.Name]; ok  {
```
ok is the conventional name for this boolean.

Reply:
Hm, I chose that because I found ok misleading -- things are not okay if that name already exists.  But shrug, I'll change it.